ui/prometheus-plugin/src/plugins/prometheus-time-series-query/query-editor-model.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useState } from 'react';
+import { TextFieldProps } from '@mui/material';
+import { produce } from 'immer';
+import { OptionsEditorProps } from '@perses-dev/plugin-system';
+import { PrometheusTimeSeriesQuerySpec } from './time-series-query-model';
+
+export type PrometheusTimeSeriesQueryEditorProps = OptionsEditorProps<PrometheusTimeSeriesQuerySpec>;
+
+/**
+ * A hook for managing the `query` state in PrometheusTimeSeriesQuerySpec. Returns the `query` value, along with
+ * `onChange` and `onBlur` event handlers to the input. Keeps a local copy of the user's input and only syncs those
+ * changes with the overall spec value once the input is blurred to prevent re-running queries in the panel's preview
+ * every time the user types.
+ */
+export function useQueryState(props: PrometheusTimeSeriesQueryEditorProps) {
```

Comment:
This file has the really the interesting changes in it. The rest of the PR changes were mostly just moving the `PrometheusTimeSeriesQuery` plugin into its own folder.